connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestServer.java

```diff
@@ -198,13 +223,39 @@ public void initializeResources(Herder herder) {
 
         registerRestExtensions(herder, resourceConfig);
 
+        List<String> adminListeners = config.getList(WorkerConfig.ADMIN_LISTENERS_CONFIG);
+        ResourceConfig adminResourceConfig;
+        if (adminListeners == null) {
+            log.info("Adding admin resources to main listener");
+            adminResourceConfig = resourceConfig;
+            addAdminResources(adminResourceConfig);
+        } else if (adminListeners.size() > 0) {
+            // TODO: we need to check if these listeners are same as 'listeners'
+            // TODO: the following code assumes that they are different
+            log.info("Adding admin resources to admin listener");
+            adminResourceConfig = new ResourceConfig();
+            adminResourceConfig.register(new JacksonJsonProvider());
+            addAdminResources(adminResourceConfig);
```

Comment:
Why have an `addAdminResources(...)` method when it could simply call `register(...)` like the previous and next lines? The method is private, so it's not useful for subclasses.

I understand that we'd be duplicating some code, but it seems more complex to have the method abstract the one call.